rust-symcrypt/src/lib.rs

```diff
@@ -90,9 +10,8 @@ pub mod hash;
 pub mod hmac;
 pub mod rsa;
 
-/// `symcrypt_init()` must be called before any other function in the library. `symcrypt_init()` can be called multiple times,
-///  all subsequent calls will be no-ops
-pub fn symcrypt_init() {
+// symcrypt_init must be called before any other API can be called. All subsequent calls to symcrypt_init will be no-ops
+fn symcrypt_init() {
```

Comment:
I was trying to find a better way to do this, because having to call `symcrypt_init` for every public function is very error prone. But it seems like there really isn't one built into the language, which seems a little crazy to me.

There is a crate that provides this functionality: [https://docs.rs/ctor/latest/ctor/](https://docs.rs/ctor/latest/ctor/)

However, it looks like the implementation is OS-specific so we might not want to take a dependency on that since we want to be as OS-agnostic as possible.

Reply:
I tried using `ctor` but ran into issues with getting it to compile properly. We can chat offline about this